iamctl/pkg/userStores/import.go

```diff
@@ -0,0 +1,195 @@
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package userstores
+
+import (
+	"bytes"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"mime"
+	"mime/multipart"
+	"net/http"
+	"net/textproto"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/wso2-extensions/identity-tools-cli/iamctl/pkg/utils"
+	"gopkg.in/yaml.v2"
+)
+
+func ImportAll(inputDirPath string) {
+
+	log.Println("Importing user stores...")
+	importFilePath := filepath.Join(inputDirPath, utils.USERSTORES)
+
+	var files []os.FileInfo
+	if _, err := os.Stat(importFilePath); os.IsNotExist(err) {
+		log.Println("No user stores to import.")
+	} else {
+		files, err = ioutil.ReadDir(importFilePath)
+		if err != nil {
+			log.Println("Error importing user stores: ", err)
+		}
+	}
+
+	for _, file := range files {
+		userStoreFilePath := filepath.Join(importFilePath, file.Name())
+		userStoreName := strings.TrimSuffix(file.Name(), filepath.Ext(file.Name()))
+
+		if !utils.IsResourceExcluded(userStoreName, utils.TOOL_CONFIGS.UserStoreConfigs) {
+			userStoreId, err := getUserStoreId(userStoreFilePath)
+			if err != nil {
+				log.Printf("Invalid file configurations for user store: %s. %s", userStoreName, err)
+			} else {
+				err := importUserStore(userStoreId, userStoreFilePath)
+				if err != nil {
+					log.Println("Error importing user store: ", err)
+				}
+			}
+		}
+	}
+}
+
+func importUserStore(userStoreId string, importFilePath string) error {
+
+	fileBytes, err := ioutil.ReadFile(importFilePath)
+	if err != nil {
+		return fmt.Errorf("error when reading the file for user store: %s", err)
+	}
+
+	// Replace keyword placeholders in the local file according to the keyword mappings added in configs.
+	fileInfo := utils.GetFileInfo(importFilePath)
+	userStoreKeywordMapping := getUserStoreKeywordMapping(fileInfo.ResourceName)
+	modifiedFileData := utils.ReplaceKeywords(string(fileBytes), userStoreKeywordMapping)
+
+	err = sendImportRequest(userStoreId, importFilePath, modifiedFileData)
+	if err != nil {
+		return fmt.Errorf("error when importing user store: %s", err)
+	}
+	return nil
+}
+
+func sendImportRequest(userStoreId string, importFilePath string, fileData string) error {
+
+	fileInfo := utils.GetFileInfo(importFilePath)
+
+	var requestMethod, reqUrl string
+	if userStoreId != "" {
+		log.Println("Updating user store: " + fileInfo.ResourceName)
+		reqUrl = utils.SERVER_CONFIGS.ServerUrl + "/t/" + utils.SERVER_CONFIGS.TenantDomain + "/api/server/v1/userstores/" + userStoreId + "/import"
+		requestMethod = "PUT"
+	} else {
+		log.Println("Creating new user store: " + fileInfo.ResourceName)
+		reqUrl = utils.SERVER_CONFIGS.ServerUrl + "/t/" + utils.SERVER_CONFIGS.TenantDomain + "/api/server/v1/userstores/import"
+		requestMethod = "POST"
+	}
```

Comment:
IMO, this control logic should be in the `ImportAll` function. we can have two separate functions for updating and creating a new user store. The idea is, a function should have a specific purpose. Hence the reason I think the control flow logic should go to the main `ImportAll` function. wdyt?

Reply:
Added 2 new functions `sendImportRequest` and `sendUpdateRequest`. The control flow logic will be at the `importUserStore` function which is called by the `ImportAll` function. According to tool's perspective, the configs to be imported will be loaded from the files, processed and sent through the relevant api call, inside the `importUserStore` method.